### PR TITLE
[Giuseppe Picciuca] functors_task_1

### PR DESCRIPTION
**Pull request by gpicciuca**
All tests pass.

### CODE REVIEW

File: functors_task_1/student/include/impl.h [hunk 1]
```diff
@@ -40,9 +51,17 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcDrink, NonAlcoholDrink nonAlcDrink)
     {
+        if (alcDrink == AlcoholDrink::Gin && nonAlcDrink == NonAlcoholDrink::LimeJuice)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: _That's a question of style, but prefer always using {} with statements_ https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es85-make-empty-statements-visible

File: functors_task_1/student/include/impl.h [hunk 2]
```diff
@@ -16,16 +25,18 @@ struct BeerOrganizer
...
 {
+    return brand == BeerBrand::Corona || brand == BeerBrand::HoeGaarden;
 }
 
 /**
  * @todo Implement lambda beer country equality comparator
  * @return true if beer county is the same, false otherwise
  */
-auto sameCountry = [](/**???*/)
+auto sameCountry = [](BeerBrand b1, BeerBrand b2)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: _Prefer making params const by default, as modifying function input params is a bad pratice_

File: functors_task_1/student/include/impl.h (hunk 1 above)
Comment (smay1613) [reply functors_task_1/student/include/impl.h]: _Also, for this task it's okay, but consider following recommendation:_
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice

**Comment by gpicciuca** [comment]
I applied the suggestions you listed above. However, one question arises in the case of the switch-statement in the mix function: Depending on how the compiler is configured, it may complain about missing default-clauses in switches. As in this case, I could have avoided adding the default-clause. Would it been ok or bad-practice?

**Pull request closed without merging**